backend/corpora/common/entities/asset.py

```diff
@@ -0,0 +1,28 @@
+import typing
+
+from backend.corpora.common.utils.s3_utils import head_file, generate_file_url
+from .entity import Entity
+from ..corpora_orm import DbDatasetArtifact
+import boto3
+
+
+class Asset(Entity):
+    table = DbDatasetArtifact
+    s3 = boto3.client("s3")
+
+    def __init__(self, db_object: DbDatasetArtifact):
+        super().__init__(db_object)
+        self.bucket_name, self.file_prefix = self.s3_uri[5:].split("/", 1)
+
+    def generate_file_url(self, expiration: int = 3600) -> typing.Union[str, None]:
+        return generate_file_url(self.bucket_name, self.file_prefix, expiration, self.s3)
```

Comment:
Why not just add the code here? `generate_file_url` is not used anywhere else in the code and is very DatasetAsset specific for now. If other places need to use that code later, then we can refactor it. Otherwise, I'd prefer to keep it simple.

Ditto for `get_s3_metadata`.

Reply:
I gave it a try and the refactors added more complexity to the tests for dataset_asset class, compared to having them split apart. I chose to favor test simplicity. Mainly I would have to add code to generate assets for testing, which doesn't exist. Adding this additional code wouldn't improve testing or be reused anywhere in the code. So I decided it would be better to keep it as is.

 I also like keeping the error handling for failed s3 request separate from the rest of the server code.

Reply:
Like you said we may use it elsewhere later... maybe. While this isn't the case presently, having it separated makes unit testing easier. So why not reap the benefits of having it split now. It saves us time, code complexity, and it's already written.

Reply:
If you really think this will make things better I can do it, but I don't agree. If it's passing around the s3 client that is the problem, we can add an s3 client to the config and import it from there.

Reply:
Can you tell me more about the complexity of adding code to generate assets for testing? It sounds like something we would actually _want_ to have.

The problem with "refactoring" prematurely is one of obfuscation. The act of generating a presigned URL and getting file size is DatasetAsset specific, not S3 specific. Because of this I state that the functions should belong in DatasetAsset because then the test would specifically test generating a presigned URL for a specific DatasetAsset. That's what we want to test is working.. not whether the boto3 is fine. 

Now on the other hand, if you were telling me that you wanted to write a function that acts upon many different kinds of objects (sanitization functions are a good example) then the functions are static and object-agnostic. This is what warrants a utility function and then DRYs code. 

All of that to say, I'd recommend combining for the following reasons:
1. No other object-agnostic use cases for the functions in the s3_utils except that it specifically acts upon DatasetAsset objects.
2. Adds yet another file to dig into to understand why a presigned URL might not be generating correcting for a DatasetAsset if we need to debug.
3. We actually want to test the functionality of generating a presigned URL for a DatasetAsset object specifically. Not arbitrarily test a presigned url generation because that provides no value.
4. Instantiating the S3 client in the s3_utils is expensive. It almost guarantees roughly a 6 second latency every time a user wants to download a dataset since the frontend call the API 3 times and each API call instantiates it twice. (roughly 1 second instantiation time last time I checked).